sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -48,6 +68,32 @@ def get_token(
         ...
 
 
+@runtime_checkable
+class ExpandedTokenCredential(TokenCredential, Protocol):
+
+    def get_expanded_token(
+        self,
+        *scopes: str,
+        claims: Optional[str] = None,
+        additional_options: Optional[Dict[str, Any]] = None,
```

Comment:
Opinions on wrapping all optional parameters into a TypedDict like so?

```python
class TokenRequestOptions(TypedDict, total=False):
    """Optional options to use when requesting an expanded token."""

    claims: str
    """Additional claims required in the token, such as those returned in a resource provider's claims
    challenge following an authorization failure."""
    tenant_id: str
    """The tenant ID to include in the token request."""
    enable_cae: bool
    """Indicates whether to enable Continuous Access Evaluation (CAE) for the requested token."""

```

Then can just have the signature be something like:
```python
def get_expanded_token(self, *scopes, options: Optional[TokenRequestOptions] = None, **kwargs)
```

Reply:
What is an implementor of the protocol expected to do if one or more additional_options are unknown?

Reply:
With a TypedDict maintaining the options, users will get typing errors when trying to pass in unknown options. However, since it's still functionally possible, I've update the docstrings of the protocol methods to explicitly stated that unknown options will be ignored.

Similarly, ignoring certain options is the current pattern with `get_token` since not all credentials support all options.